DangProject/Sources/Domain Layer/UseCases/DefaultFetchProfileUseCase.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  DefaultFetchProfileUseCase.swift
+//  DangProject
+//
+//  Created by 김동우 on 2022/07/21.
+//
+
+import UIKit
+
+import RxSwift
+
+class DefaultFetchProfileUseCase: FetchProfileUseCase {
+    private let disposeBag = DisposeBag()
+    private var coreDataManagerRepository: CoreDataManagerRepository
+    private let firebaseFireStoreUseCase: FirebaseFireStoreUseCase
+    private let firebaseStorageUseCase: FirebaseStorageUseCase
+    
+    init(coreDataManagerRepository: CoreDataManagerRepository,
+         firebaseFireStoreUseCase: FirebaseFireStoreUseCase,
+         firebaseStorageUseCase: FirebaseStorageUseCase) {
+        self.coreDataManagerRepository = coreDataManagerRepository
+        self.firebaseFireStoreUseCase = firebaseFireStoreUseCase
+        self.firebaseStorageUseCase = firebaseStorageUseCase
+    }
+    
+    func fetchProfileData() -> Observable<ProfileDomainModel> {
+        return Observable.create { [weak self] emitter in
+            guard let strongSelf = self else { return Disposables.create() }
+            if UserDefaults.standard.bool(forKey: UserInfoKey.isLatestProfileData) != true {
+                strongSelf.fetchRemoteProfileData()
+                    .subscribe(onNext: { profileData in
+                        emitter.onNext(profileData)
+                    })
+                    .disposed(by: strongSelf.disposeBag)
+                UserInfoKey.setIsLatestProfileData(true)
+            } else {
+                strongSelf.fetchLocalProfileData()
+                    .subscribe(onNext: { profileData in
+                        emitter.onNext(profileData)
+                    })
+                    .disposed(by: strongSelf.disposeBag)
+            }
```

Comment:
[DangProject/Sources/Application/AppCoordinator.swift](https://github.com/SamsungTech/DangProject/pull/20/files/6fa45cbd7d20fd0153c0be1e17c6b1b3afcfd638#diff-c7399f6c8ea380487462d0615e5da30bf3d5cbefad3434ae7800a75bccb4c79b)
관련 모델 struct 내에 타입 프로퍼티로 저장 되어 있다는 가정 하에
```suggestion
            if ProfileDomainModel.isLatestProfileData {
                strongSelf.fetchLocalProfileData()
                    .subscribe(onNext: { profileData in
                        emitter.onNext(profileData)
                    })
                    .disposed(by: strongSelf.disposeBag)
            } else {
                strongSelf.fetchRemoteProfileData()
                    .subscribe(onNext: { profileData in
                        emitter.onNext(profileData)
                    })
                    .disposed(by: strongSelf.disposeBag)
                ProfileDomainModel.isLatest = true
            }
```
이 방식이 훨씬 가독성이 좋을 것 같습니다.

Reply:
00a81fc032c3c3aa453607a61f4bd21884c7a612
했습니다.

Reply:
d26b37ca6469b5791a8d5fcea1afccc0b9091144
수정했습니다.